src/components/profile/TableNotifications.vue

```diff
@@ -0,0 +1,184 @@
+        label: i18n.global.t('notifications.labelDate'),
+        field: 'timestamp',
+        format: (val: number | string | null): string =>
+          val ? date.formatDate(new Date(String(val)), 'D. MMM. YYYY') : '',
+        sortable: true,
+        required: true,
+      },
+      {
+        name: 'unread',
+        align: 'left',
+        label: i18n.global.t('notifications.labelState'),
+        field: 'unread',
+        format: (val: number | string | null): string =>
+          val
+            ? i18n.global.t('notifications.labelUnread')
+            : i18n.global.t('notifications.labelRead'),
+        sortable: true,
+        required: true,
+      },
+      {
+        name: 'action',
+        align: 'right',
+        label: i18n.global.t('notifications.labelAction'),
+        field: 'action',
+        sortable: false,
+        required: false,
+      },
+    ];
+
+    const rows = reactive<Notification[]>(tableNotificationsFixture);
+
+    const markAsRead = (row: Notification): void => {
+      row.unread = false;
+    };
+
+    const markAllAsRead = (): void => {
+      rows.forEach((row) => {
+        markAsRead(row);
+      });
+    };
+
+    const unreadCount = computed(() => {
+      return rows.filter((row) => row.unread).length;
+    });
+
+    return {
+      columns,
+      rows,
+      markAsRead,
+      markAllAsRead,
+      unreadCount,
+    };
+  },
+});
+</script>
+
+<template>
+  <div data-cy="table-notifications">
+    <div class="flex justify-end q-mb-lg">
+      <q-btn
+        unelevated
+        rounded
+        outline
+        color="primary"
+        :disabled="unreadCount === 0"
+        data-cy="button-mark-all-as-read"
+        @click.prevent="markAllAsRead"
+      >
+        <q-icon name="check" size="18px" class="q-mr-sm" />
+        {{ $t('notifications.labelMarkAllAsRead') }}
+      </q-btn>
+    </div>
+
+    <!-- Table -->
+    <q-table flat bordered :rows="rows" :columns="columns" row-key="title">
+      <template v-slot:body="props">
+        <q-tr :props="props" data-cy="notification-row">
+          <q-td
+            key="title"
+            :props="props"
+            class="text-grey-10"
+            data-cy="notification-title"
+          >
+            <q-icon
+              :name="props.row.data.icon"
+              size="18px"
+              color="primary"
+              class="q-mr-sm"
+              data-cy="notification-icon"
+            />
+            <component
+              :is="props.row.data.url ? 'a' : 'span'"
+              :href="props.row.data.url"
+              :class="props.row.unread ? 'text-weight-bold' : ''"
+              target="_blank"
+              data-cy="notification-verbal"
+              >{{ props.row.verb }}</component
+            >
+          </q-td>
+          <q-td key="timestamp" :props="props" data-cy="notification-timestamp">
+            <template v-for="col in props.cols" :key="col.field">
+              <span v-if="col.field === 'timestamp'">
```

Comment:
You can use TS Enum for `'timestamp'` literal.

Reply:
Refactored into enum.